src/config.rs

```diff
@@ -1,80 +1,127 @@
+    pub perf: PerfConfig,
+}
 
-    pub fn num_threads(&self) -> usize {
-        match self {
-            Self::Sequential => 1,
-            Self::Parallel => num_cpus::get(),
-            Self::ParallelN(threads) => *threads,
-        }
+impl Settings {
+    /// Loads config settings from a file or env vars
+    /// The public parameter disk cache can also be overriden by a CLI arg if applicable
+    pub fn from_config(
+        config_file: &Utf8PathBuf,
+        settings: Option<&HashMap<&str, String>>,
+    ) -> Result<Self, ConfigError> {
+        let public_params = "public_params_dir";
+        // Settings are read first to last, in order of increasing precedence.
+        // Hence, default values must come first so they are overriden by all other methods.
+        Config::builder()
+            // Default settings if unspecified in the config file
+            .set_default(public_params, public_params_default_dir().to_string())?
+            .set_default("perf", "fully-sequential".to_string())?
+            .add_source(File::with_name(config_file.as_str()).required(false))
+            // Then override with any `LURK` environment variables
+            .add_source(Environment::with_prefix("LURK"))
+            // Optionally override if settings were specified via CLI arg
+            .set_override_option(public_params, settings.and_then(|s| s.get(public_params).map(|v| v.to_owned())))?
+            .build()
+            .and_then(|c| c.try_deserialize())
     }
+}
 
-    pub fn chunk_size(&self, total_n: usize, min_chunk_size: usize) -> usize {
-        if self.is_sequential() {
-            total_n
-        } else {
-            let num_threads = self.num_threads();
-            let divides_evenly = total_n % num_threads == 0;
-
-            ((total_n / num_threads) + usize::from(!divides_evenly)).max(min_chunk_size)
+impl Default for Settings {
+    fn default() -> Self {
+        Self {
+            public_params_dir: public_params_default_dir(),
+            perf: PerfConfig::default(),
         }
     }
 }
 
-#[derive(Default, Debug)]
-pub struct ParallelConfig {
-    pub recursive_steps: Flow,    // Multiple `StepCircuit`s.
-    pub synthesis: Flow,          // Synthesis (within one `StepCircuit`)
-    pub poseidon_witnesses: Flow, // The poseidon witness part of synthesis.
+pub fn public_params_default_dir() -> Utf8PathBuf {
+    #[cfg(not(target_arch = "wasm32"))]
+    let params_path = home_dir();
+    #[cfg(target_arch = "wasm32")]
+    let params_path = Utf8PathBuf::new();
+    params_path.join(".lurk/public_params")
 }
 
-/// Should we use optimized witness-generation when possible?
-#[derive(Debug, Default)]
-pub struct WitnessGeneration {
-    // NOTE: Neptune itself *will* do this transparently at the level of individual hashes, where possible.
-    // so this configuration is only required for higher-level decisions.
-    pub precompute_neptune: bool,
+// TODO: Should we crash if the user has no home dir?
```

Comment:
I think the crash is fine for now, since we exclude wasm. We'll eventually want to be a bit more clever later, when our config is sufficiently refined.